src/store-payload.js

```diff
@@ -0,0 +1,96 @@
+// store-payload.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+/**
+ * Stores the given payload for later use or returns a hash of the payloads by
+ * port names. Each payload will have a unique index by vnid. If a port is
+ * addressable then the payload from each socket is stored separately and an
+ * Array of payloads is returned for those ports. Omitting the payload will
+ * not change anything, but return a hash of payloads by port name. Passing an
+ * undefined payload will remove any existing the indexed payload entry.
```

Comment:
Overall, I like the idea of using socketIndex instead of sourceNodeName and sourcePortName to index the inputStates.  That seems easier and more efficient, and we can address the issue of making them predictable when the NoFlo graph is reloaded or reconfigured.  However, I would feel more comfortable if: 1. The data were stored on a per-VNI basis, rather than per-port.  I'm not seeing a real benefit to storing the data on the port, and it fragments the VNI object.  2. The entire inputStates object were created at once when the VNI is created (createInputStates), rather than being created incrementally as events come in on different ports.